autoload/mucomplete.vim

```diff
@@ -269,6 +271,30 @@ fun! s:extend_completion(dir, keys)
         \ : a:keys
 endf
 
+fun! s:get_chain_for_buffer()
+    return get(b:, 'mucomplete_chain',
+          \ get(g:mucomplete#chains, getbufvar("%", "&ft"),
+          \     g:mucomplete#chains['default']))
+endfun
+
+fun! s:get_chain_for_scope()
+  if !has_key(g:mucomplete#scoped_chains, &ft)
+    " no scoped chains for this filetype
+    return s:get_chain_for_buffer()
+  endif
+
+  let scope = synIDattr(synID(line('.'), col('.')-1, 1), 'name')
```

Comment:
Is there a specific reason to set the third argument of `synIDattr()` to `1`? According to Vim's manual, when setting it to `0`,

>[…] the transparent item is returned. This is useful when wanting to know which syntax item is effective (e.g., inside parens).

It seems to me that the use we make of the syntax ID matches the above description well.

Reply:
Could be, I've been using that snippet as I found it in VimWikia (or somewhere else), to be honest.